### ISSUE TITLE
Can't restart websockets on Linux, but can on Windows

### ISSUE THREAD

**Issue by simaster123** [sammchardy/python-binance#409]
I've been having issues reconnecting to the Websockets after a dropout; similar to what's described here:  https://github.com/sammchardy/python-binance/issues/202

I implemented code similar to what @sammchardy recommended in the comments, which works well when I run it on my Windows 10 machine.  However, when I port the code to my Ubuntu 19 machine, it doesn't work.  Both are using the same versions of twisted and autobahn:

python-binance = 0.7.1
twisted = 19.2.1
autobahn = 19.7.1

Does anyone have any idea what might be the cause / where I should dig to troubleshoot?

**Comment by ftuyama** [comment]
I've tried using this approach: https://github.com/sammchardy/python-binance/issues/202#issuecomment-437614147 and the websocket reconnection works, but there's a huge delay (~30s).

I've tried also restarting reactor, but it can't be restarted https://github.com/sammchardy/python-binance/issues/45.
Killing BinanceSocketManager thread isn't a good approach either.

I can't figure out a good solution to quickly restart the websocket connection, all I can do is restart the process, which is not a good solution either.

**Comment by diazagasatya** [comment]
I have the same problem currently running on linux, however restarting websocket works well on Mac. 

python3.7
autobahn==20.4.3
Twisted==20.3.0

Any update on how to resolve this issue?

**Comment by diwenshi61** [comment]
Felt like I was taking crazy pills before I found this thread, restarting sockets worked fine on Windows, but nothing would be received after restarting on Linux. I tried multiple versions of Ubuntu, Debian, a bunch of versions of python and packages, nothing worked. The maxRetries solution doesn't seem to work for me either. This renders the entire package practically unusable on Linux in my case.

**Comment by badjano** [comment]
sockets not working on Raspbian Buster on raspberry pi either... any chance this gets fixed?

**Comment by StephanAkkerman** [comment]
I have this problem on my Raspberry pi as well, but I made a workaround. To restart my script I've used time.sleep(seconds) and then os._exit(0). In another script I've a while True loop with: "os.system("python3 FileName.py")" in it. So after a certain amount of time the script exits and then gets directly restarted again. So to have my original script running all the time, with restarts, I only need to start the script with the while True loop in it. The only problem I can think of with this solution is if you have temporary data saved in the script that exits. Hopefully my solution helps other people having this issue.
